src/constants/HIDmap.ts

```diff
@@ -729,376 +1113,334 @@ export class Hid {
+    Hid.C,//6710,
+    Hid.D,//6810,
+    Hid.E,//6910,
+    Hid.F,//7010,
+    Hid.G,//7110,
+    Hid.H,//7210,
+    Hid.I,//7310,
+    Hid.J,//7410,
+    Hid.K,//7510,
+    Hid.L,//7610,
+    Hid.M,//7710,
+    Hid.N,//7810,
+    Hid.O,//7910,
+    Hid.P,//8010,
+    Hid.Q,//8110,
+    Hid.R,//8210,
+    Hid.S,//8310,
+    Hid.T,//8410,
+    Hid.U,//8510,
+    Hid.V,//8610,
+    Hid.W,//8710,
+    Hid.X,//8810,
+    Hid.Y,//8910,
+    Hid.Z,//9010,
+    Hid.N0,//4810,
+    Hid.N1,//4910,
+    Hid.N2,//5010,
+    Hid.N3,//5110,
+    Hid.N4,//5210,
+    Hid.N5,//5310,
+    Hid.N6,//5410,
+    Hid.N7,//5510,
+    Hid.N8,//5610,
+    Hid.N9,//5710,
+    Hid.NP0,//9610,
+    Hid.NP1,//9710,
+    Hid.NP2,//9810,
+    Hid.NP3,//9910,
+    Hid.NP4,//10010,
+    Hid.NP5,//10110,
+    Hid.NP6,//10210,
+    Hid.NP7,//10310,
+    Hid.NP8,//10410,
+    Hid.NP9,//10510,
+    Hid.ENTER,//1310,
+    Hid.ESCAPE,//2710,
+    Hid.BACKSPACE,//810,
+    Hid.TAB,//910,
+    Hid.SPACE,//3210,
+    Hid.MINUS,//18910,
+    Hid.EQUAL,//18710,
+    Hid.BRACKETL,//21910,
+    Hid.BRACKETR,//22110,
+    Hid.BACKSLASH,//22010,
+    Hid.SEMICOLON,//18610,
+    Hid.QUOTE,//22210,
+    Hid.BACKQUOTE,//19210,
+    Hid.COMMA,//7710,
+    Hid.PERIOD,//18810,
+    Hid.CAPSLOCK,//2010,
+    Hid.SLASH,//19110,
+    Hid.F1,//11210,
+    Hid.F2,//11310,
+    Hid.F3,//11410,
+    Hid.F4,//11510,
+    Hid.F5,//11610,
+    Hid.F6,//11710,
+    Hid.F7,//11810,
+    Hid.F8,//11910,
+    Hid.F9,//12010,
+    Hid.F10,//12110,
+    Hid.F11,//12210,
+    Hid.F12,//12310,
+    Hid.F13,//12410,
+    Hid.F14,//12510,
+    Hid.F15,//12610,
+    Hid.F16,//12710,
+    Hid.F17,//12810,
+    Hid.F18,//12910,
+    Hid.F19,//13010,
+    Hid.F20,//13110,
+    Hid.F21,//13210,
+    Hid.F22,//13310,
+    Hid.F23,//13410,
+    Hid.F24,//13510,
+    Hid.PRINTSCREEN,//4410,
+    Hid.SCROLLLOCK,//14510,
+    Hid.PAUSE,//1910,
+    Hid.INSERT,//4510,
+    Hid.HOME,//3610,
+    Hid.PAGEUP,//3310,
+    Hid.DELETE,//4610,
+    Hid.END,//3510,
+    Hid.PAGEDOWN,//3410,
+    Hid.ARROWR,//3910,
+    Hid.ARROWL,//3710,
+    Hid.ARROWD,//4010,
+    Hid.ARROWU,//3810,
+    Hid.NUMLOCK,//14410,
+    Hid.NUMDIVIDE,//11110,
+    Hid.NUMMULTIPLY,//10610,
+    Hid.NUMSUBTRACT,//10910,
+    Hid.NUMADD,//10710,
+    Hid.NUMENTER,//1310,
+    Hid.NUMDECIMAL,//11010,
+//
+    Hid.SHIFTL,//1611,
+    Hid.CONTROLL,//1711,
+    Hid.ALTL,//1811,
+    Hid.METAL,//9111,
+    Hid.SHIFTR,//1612,
+    Hid.CONTROLR,//1712,
+    Hid.METAR,//9212,
+    Hid.ALTR//1812,
   ]
 }
 
-export const webCodeHIDLookup = new Map<string, HidInfo>(
+export const webCodeLocationHIDLookup= new Map<string, HidInfo>(
+  // This creates a unique ID: whichID, separated by an extra '1' digit, then locationID.
   Hid.all
-    .filter((hid) => hid.webKeyId !== undefined)
-    .map((hid) => [hid.webKeyId!, hid])
+    .filter((hid) => hid.whichID !== undefined)
+    .map((hid) => [hid.whichID!.toString() + '1' + hid.locationID!.toString(), hid])
```

Comment:
Function now returns the combined ID of the key with position

Reply:
With the code right now, the !'s are completely redundant, as these values are always defined

Reply:
also, why use `'1'` as the separator and not something like `|`, `/`, `,` which would make it more clear that it's an encoded value?

Reply:
Using 1 also makes it a pain in the annoying to decode this, as you can't just do `.split(...)` as 1 can appear in each of the values you're encoding

Reply:
Swapped: [1d0ecb3](https://github.com/WootingKb/wooting-macros/pull/194/commits/1d0ecb3ccb4153c8d55321bcdedf6adc341ad782)